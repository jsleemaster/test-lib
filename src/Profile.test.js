import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username="유저이름" name="테스트" />);
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the props correctly', () => {
    const utils = render(<Profile username="유저이름" name="테스트" />);
    utils.getByText('유저이름'); // 유저이름이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText(/테스트/); // 정규식 // 을 통과하는 엘리먼트가 있는지 확인
  });
});