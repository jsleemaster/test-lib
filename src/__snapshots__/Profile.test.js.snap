// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      유저이름
    </b>
     
    <span>
      (
      테스트
      )
    </span>
  </div>
</div>
`;
